src/NLog/NLogTraceListener.cs

```diff
@@ -457,6 +430,37 @@ protected virtual void ProcessLogEventInfo(LogLevel logLevel, string loggerName,
             logger.Log(ev);
         }
 
+        private ILogger GetLogger(string loggerName, out StackTrace stackTrace, out int userFrameIndex)
+        {
+            loggerName = (loggerName ?? Name) ?? string.Empty;
+
+            stackTrace = null;
+            userFrameIndex = -1;
+            if (AutoLoggerName)
+            {
+                stackTrace = new StackTrace();
```

Comment:
Double check, it isn't a problem that `new StackTrace` has been moved to an inner method?

Reply:
Should not have any issues. But just to be sure, then I have moved capture of StackTrace back to original location.